File: 프로그래머스/0/120843. 공 던지기/공 던지기.js
function solution(numbers, k) {
  let arr = [];
  let result = [];
  const arrCount = Math.ceil((k * 2) / numbers.length)
  for (let i = 0; i < arrCount; i++) {
    arr.push(...numbers)
  }
  for (let j = 0; j < arr.length; j+=2) {
    result.push(arr[j])
  }
  return result[k-1]
}

File: 프로그래머스/0/120843. 공 던지기/공 던지기.java
class Solution {
        public int solution(int[] numbers, int k) {
        int index = 0;
            
        if (k * 2 <= numbers.length) {
            for (int i = 1; i < k; i++) {
                index = (index + 2) % numbers.length;
            }
        } else {
            int[] extendedNumbers = new int[numbers.length * 2];
            System.arraycopy(numbers, 0, extendedNumbers, 0, numbers.length);
            System.arraycopy(numbers, 0, extendedNumbers, numbers.length, numbers.length);

            for (int i = 1; i < k; i++) {
                index = (index + 2) % extendedNumbers.length;
            }
            return extendedNumbers[index];
        }
        
        return numbers[index];
    }
}